src/lib.rs

```diff
@@ -357,6 +387,73 @@ impl PrometheusMetricsBuilder {
     }
 }
 
+#[derive(Debug)]
+/// Configuration for a single metric
+///
+/// Allows configuring name and labels set for the metric
+pub struct ActixMetric {
+    name: String,
+    labels: Vec<String>,
+}
+
+impl ActixMetric {
+    /// Create a new metric configuration
+    pub fn new(name: &str, labels: Vec<&str>) -> ActixMetric {
+        ActixMetric {
+            name: name.to_string(),
+            labels: labels.into_iter().map(|s| s.to_string()).collect(),
+        }
+    }
+}
+
+#[derive(Debug)]
+/// Configuration for the collected metrics
+///
+/// Stores individual metric configuration objects
+pub struct ActixMetricsConfiguration {
+    http_requests_total: ActixMetric,
+    http_requests_duration_seconds: ActixMetric,
+}
+
+impl ActixMetricsConfiguration {
+    /// Create the default metrics configuration
+    fn default() -> ActixMetricsConfiguration {
+        ActixMetricsConfiguration {
+            http_requests_total: ActixMetric::new(
+                "http_requests_total",
+                vec!["endpoint", "method", "status"],
+            ),
+            http_requests_duration_seconds: ActixMetric::new(
+                "http_requests_duration_seconds",
+                vec!["endpoint", "method", "status"],
+            ),
+        }
+    }
+
+    /// Create new metrics configuration with given input
```

Comment:
I think this seems extraneous?

Arguably it would be fine to not have this self constraining `new` of two initial parameters and instead simply do:

```rust
ActixMetricsConfiguration::default().
  http_requests_total(http_requests_total).
  http_requests_duration_seconds(http_requests_duration_seconds)
```

That way, if the `ActixMetricsConfiguration` struct ever changes, we don't need to do a breaking change on `new`.

Reply:
My rationale for adding `fn new` was that it's easier and more logical way of initializing things, but you're right, any change to the struct would be breaking. I'll remove the `new`.

Reply:
My rationale for adding `fn new` was that it's easier and more logical way of initializing things, but you're right, any change to the struct would be breaking. Using the default to initialize in the first place guides the user to a better direction - to custom configure only the metrics that absolutely need that.
I'll remove the `new`.